src/main/java/moze_intel/projecte/gameObjs/container/slots/transmutation/SlotInput.java

```diff
@@ -38,7 +38,7 @@ public void putStack(@Nonnull ItemStack stack)
 		if (stack.getItem() instanceof IItemEmc)
 		{
 			IItemEmc itemEmc = ((IItemEmc) stack.getItem());
-			double remainingEmc = itemEmc.getMaximumEmc(stack) - (int) Math.ceil(itemEmc.getStoredEmc(stack));
+			long remainingEmc = itemEmc.getMaximumEmc(stack) - (int) Math.ceil(itemEmc.getStoredEmc(stack));
```

Comment:
casting to int here feels wrong?

Reply:
Yeah it does, I'll probably change that to long